File: Applied Course/4.Problem Solving/7.Problems on Strings/10.Reorder Data In log files.cpp
#include<bits/stdc++.h>
using namespace std;
class Solution {
public:
    vector<string> reorderLogFiles(vector<string>& logs) {
        vector<string> let,dig;
        for(auto& i:logs) {
            if(isalpha(i[i.find(' ')+1])) 
                let.push_back(i);
            else 
                dig.push_back(i);
        }
        sort(let.begin(),let.end(),[](const string& s1, const string& s2) {
            size_t p1=s1.find(' ')+1, p2=s2.find(' ')+1;
            int res=s1.substr(p1).compare(s2.substr(p2));
            if(res<0) 
                return true;
            else if(res>0) 
                return false;
            return s1.substr(0,p1-1).compare(s2.substr(0,p2-1)) < 0;
        });
        int pos=-1;
        for(auto& i:let) 
            logs[++pos]=i;
        for(auto& i:dig) 
            logs[++pos]=i;
        return logs;
    }
};
int main()
{
    // https://leetcode.com/problems/reorder-data-in-log-files
    return 0;
}

File: Applied Course/4.Problem Solving/7.Problems on Strings/10.Reorder Data In log files.py
class Solution:
    def reorderLogFiles(self, logs: List[str]) -> List[str]:
        digit_logs,letter_logs=[],[]
        for log in logs:
            second_word=log.split()[1]
            if second_word.isdigit():
                digit_logs.append(log)
            else:
                letter_logs.append(log)
        letter_logs.sort(key=lambda x:x.split()[0])
        letter_logs.sort(key=lambda x:x.split()[1:])
        return letter_logs+digit_logs